src/libraries/System.Memory/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -476,10 +477,31 @@ private static unsafe void Avx2Decode(ref byte* srcBytes, ref byte* destBytes, b
             destBytes = dest;
         }
 
+        // This can be replaced once https://github.com/dotnet/runtime/issues/63331 is implemented.
+        // Note that for performance reasons this is not functionally identical on SSE3 and Arm64. On both, the input
+        // right is truncated to 4bits per lane. In addition on SSE3, the shuffle will return 0 for each input in right
+        // where the top bit is set. On Arm64 the top bit will be ignored.
```

Comment:
This could be the same if the mask was different. `VectorTableLookup` works as:
> If an index is out of range for the table, the result for that lookup is 0.

Since you're masking anyways, you could just mask to `0x8F` (keeping the most significant bit and the 4 least significant bits).